### PR TITLE
Space-Travelers-Hub: Overview

### PR DESCRIPTION
**Pull request by Kwaku28**
### Overview

- [ ] The Rockets section displays a list of all available SpaceX rockets. Users can book each rocket by clicking the reservation button or cancel the previously made booking.
- [ ] The Missions section displays a list of current missions along with their brief description and participation status. There is also a button next to each mission that allows users to join the selected mission or leave the mission the user joined earlier.
- [ ] The My Profile section displays all reserved rockets and space missions.

### 1. Configuration & basic setup

- [ ] Create React App.
- [ ] Install [React Redux](https://react-redux.js.org/), [Redux Logger](https://www.npmjs.com/package/redux-logger) and [React Router](https://github.com/microverseinc/curriculum-react-redux/blob/main/math-magicians/articles/react_router.md).
- [ ] Downloaded the [free image](https://www.flaticon.com/free-icon/planet_3212567?term=space&page=1&position=19&page=1&position=19&related_id=3212567&origin=style) for the app logo.
- [ ] Created routes and view components: rockets, missions, my profile.
 Used `<NavLink />` for the page navigation links and style active class (underline active navigation link) to indicate which section/page the user is currently on.
- [ ] Created directories for all Redux state slice files: `rockets` and `missions`.

### 2. Redux and Render UI

- [ ] Fetched data upon render from the SpaceX API endpoints.
- [ ] Dispatched an action to store the selected data in the Redux store.
- [ ] Used `useSelector()` Redux Hook to select the state slices and render lists of rockets and missions in corresponding routes
- [ ] Rendered a list of rockets.
- [ ] Rendered a table with the missions' data.
- [ ] When a user clicks the "Reserve rocket" button, an action is dispatched to update the store.
- [ ] When a user clicks the "Join Mission" button, an action is dispatched to update the store.
- [ ] Rockets that have already been reserved show a "Reserved" badge and a "Cancel reservation" button instead of the default "Reserve rocket".
- [ ] Missions that the user has joined already show a badge "Active Member" instead of the default "NOT A MEMBER" and a button "Leave Mission" instead of the "Join Mission" button.
- [ ] Used `filter()` to render a list of all joined missions and reserved rockets on the "MyProfile" page.

Thank you🤝

### CODE REVIEW

File: src/Redux/missions/missionsSlice.js [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+};
+
+const URL = 'https://api.spacexdata.com/v3/missions';
+
+export const fetchMissions = createAsyncThunk(
+  'missions/fetchMissions',
+  async () => {
+    try {
+      const response = await axios.get(URL);
+      return response.data;
```
Comment (divinecharlotte) [inline src/Redux/missions/missionsSlice.js:23]: - [ ] Kindly consider dispatching an action  to store the selected data  which are 

```
* mission_id
* mission_name
* description
```

in Redux store
doing so will help you to meet the below requirement
Screen Shot 2023-05-19 at 14 16 35

File: src/components/activeMission.js [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+  useEffect(() => {
+    dispatch(bookedMissions());
+  }, [dispatch]);
+
+  return (
+    <>
+      <div className={styles.container}>
+        <h2 className={styles.header}>My Missions</h2>
+        {reservedMission.reservedMission?.map((item) => (
+          <ul key={item.mission_id} className={styles.missionlist}>
```
Comment (divinecharlotte) [inline src/components/activeMission.js:19]: - [ ] kindly consider rendering  a list of all joined missions using `filter()` to meet the below requirement 
> Render a list of all joined missions (use filter()).

**Review by divinecharlotte — CHANGES_REQUESTED** [review]
Hi @TEAM      , 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

## To Highlight 💯 

- linters are passing 👍 
- Descriptive Pr 👍 
- Descriptive commit messages 👍 



 

## Required Changes ♻️

- [ ] kindly consider writing these for all your components and functions to meet the below requirement
Screen Shot 2023-05-19 at 14 55 00


_Check the comments under the review._
## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by joel-kalema — CHANGES_REQUESTED** [review]
# STATUS: REQUIRED CHANGES :recycle::recycle: 

Hi Team
Good job so far! I know you worked hard to implement this project, I really appreciate the effort you put in! But there are some issues that you still need to work on to go to the next project but you are almost there! 

almost-there

## To highlight :champagne: 
- All linters are passing :heavy_check_mark: 
- Good commit history :heavy_check_mark: 
- Your design looks good :100: :heavy_check_mark: 
- Good PR description :heavy_check_mark: 
- Follow GitHub flow :heavy_check_mark: 

## Required changes :recycle: :recycle: 
- [ ] Check the comments under the review.

## Optional suggestions

_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

_Feel free to leave any questions or comments in the PR thread if something is not 100% clear. 
Please ping me @joel-kalema  when you comment so I can receive the notification or use slack with the same name._

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by elmar8287 — APPROVED** [review]
Hi, 

# APPROVED  👍 🥇 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 

Congratulations! 🎉 

### Highlights

- All tests passed
Screen Shot 2023-05-20 at 13 16 44


### Optional suggestions

- N.A

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._



Cheers, and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
### Please, remember to tag me (@elmar8287) in your question so I can receive the notification 👍 
giphy

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**